control-plane-operator/controllers/hostedcontrolplane/util/containers.go

```diff
@@ -11,6 +11,19 @@ func BuildContainer(container *corev1.Container, buildFn func(*corev1.Container)
 	return *container
 }
 
+// ApplyContainer will add or update container within containers and return an
+// array of containers with the mutated container.
+func ApplyContainer(containers []corev1.Container, container *corev1.Container, buildFn func(*corev1.Container)) []corev1.Container {
```

Comment:
With this we don't really need any of the other container defaulting code from the other PRs. The root of the issue with container updates was `BuildContainer` reconstructing the containers with new structs every time.